### PR TITLE
feat: adds editor functionality to CodeDisplay

### PR DESCRIPTION
**Pull request by noahamac**
Replaces `AceEditor` with `@looker/code-editor` for all code display within RunIt and ApiExplorer.

Search pattern marking:
image

Editor:
image

Response body:
image

Code tab:
image

Type declaration:
image

### CODE REVIEW

File: packages/run-it/src/components/ShowResponse/responseUtils.tsx
Comment (josephaxisa) [inline packages/run-it/src/components/ShowResponse/responseUtils.tsx:119]:
```suggestion
  <CodeDisplay language="html" code={response.body.toString()} transparent />
```

File: yarn.lock
Comment (bryans99) [inline yarn.lock:2196]: Hm, how'd this get pulled in?

File: yarn.lock
Comment (bryans99) [inline yarn.lock:2228]: Ditto

File: packages/code-display/src/CodeDisplay.tsx
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:127]: is this for when `CodeDisplay` is used as an editor? If so I think a comment would be handy

File: packages/code-display/src/CodeDisplay.tsx
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:51]: Add a comment perhaps describing what happens if no height is specified? I don't see this prop being used anywhere

File: packages/code-display/src/CodeDisplay.tsx
Comment (josephaxisa) [inline packages/code-display/src/CodeDisplay.tsx:127]: Also L117 can be modified to use the Span from @looker/components

File: packages/code-display/src/CodeDisplay.tsx
Comment (jkaster) [inline packages/code-display/src/CodeDisplay.tsx:130]: I think we should have separate components for Display and Editing. I'm presuming `Editor` is heavier than `Highlight`

**Review by jkaster — CHANGES_REQUESTED** [review]
Make CodeDisplay and CodeEditor distinct components

File: packages/code-display/src/CodeDisplay.tsx
Comment (noahamac) [reply packages/code-display/src/CodeDisplay.tsx:127]: We can't use Span as it has default styles which conflict with Prism's styles applied onto a span

**Comment by bryans99** [comment]
I removed a reference to an old looker components-test-utils from yarn.lock

**Comment by jkaster** [comment]
Noah is working on completely removing references to Ace for this PR, so we'll hold off on approval for now

**Comment by joeldodge79** [comment]
I wonder if we should just submit a separate PR to unbreak `main` first?

**Comment by jkaster** [comment]
Is main still broken?

File: packages/code-editor/src/CodeDisplay.tsx
Comment (jkaster) [inline packages/code-editor/src/CodeDisplay.tsx:96]: I don't know what the thinking is on calling util functions like this in the markdown. Probably fine. I typically establish the values as consts (or use state if they can change from user interaction) but defer to @bryans99 or @josephaxisa on this topic

File: packages/code-editor/src/utils/utils.ts
Comment (jkaster) [inline packages/code-editor/src/utils/utils.ts:61]: hugely minor point: typically you put the positive test first, then the negative goes in the else. Some linters will recommend that.

File: yarn.lock
Comment (jkaster) [inline yarn.lock:3476]: Look at those glorious deletions!

**Review by jkaster — DISMISSED** [review]
LGTM. Nice to bid Ace adieu!

File: packages/code-editor/src/CodeDisplay.tsx
Comment (noahamac) [reply packages/code-editor/src/CodeDisplay.tsx:96]: Yeah, curious to know the standard on this.

**Review by jkaster — APPROVED** [review]

**Comment by noahamac** [comment]
What a ride. Thanks for the help, everyone :smile_cat:

**Pull request merged**
